src/editor/actions/actions.js

```diff
@@ -256,80 +259,29 @@ export function getNotebookSaveRequestOptions(state, options = undefined) {
   return postRequestOptions;
 }
 
-export function saveNotebookRequest(
-  url,
-  postRequestOptions,
-  dispatch,
-  appMsg = true
-) {
-  return fetchWithCSRFTokenAndJSONContent(url, postRequestOptions)
-    .then(response => {
-      if (!response.ok) {
-        return response
-          .json()
-          .then(json => {
-            // this is a horrible and obvious hack that will go away
-            // when we don't provide an easy interface for the user
-            // to double save a revision (i.e. we have provided
-            // continuous autosave)
-            if (json.non_field_errors && json.non_field_errors.length) {
-              // no changes, not really an error, we'll silently pretend
-              // this didn't happen
-              return {};
-            }
-            // else, some kind of genuine error
-            dispatch(
-              updateAppMessages({
-                message: "Error saving notebook.",
-                messageType: "ERROR_SAVING_NOTEBOOK"
-              })
-            );
-            throw response;
-          })
-          .catch(() => {
-            throw response;
-          });
-      }
-      return response.json();
-    })
-    .catch(err => {
-      if (appMsg) {
-        dispatch(
-          updateAppMessages({
-            message: `Error Saving Notebook.`,
-            messageType: "ERROR_SAVING_NOTEBOOK"
-          })
-        );
-      }
-      throw new Error(err.message);
-    });
-}
-
 export function createNewNotebookOnServer(options = { forkedFrom: undefined }) {
   return (dispatch, getState) => {
     const state = getState();
-    const postRequestOptions = getNotebookSaveRequestOptions(state, {
-      forkedFrom: options.forkedFrom
-    });
-    return saveNotebookRequest(
-      "/api/v1/notebooks/",
-      postRequestOptions,
-      dispatch
-    )
-      .then(json => {
+    return createNotebookRequest(state.title, state.jsmd, options)
+      .then(notebook => {
```

Comment:
plz refactor this as async/await

Reply:
We shouldn't do this, as the function calling this is expecting a promise as a return value.

Reply:
async functions always return a Promise, and if there is no return value, I believe it will return `undefined`

Reply:
to pass eslint I needed to make these functions return a dummy value of undefined. Not so happy with that, but it seems to work